shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-common/src/main/java/org/apache/shardingsphere/sharding/converter/ShardingRuleStatementConverter.java

```diff
@@ -46,19 +47,35 @@
      * @return YAML sharding rule configuration
      */
     public static YamlShardingRuleConfiguration convert(final CreateShardingRuleStatement sqlStatement) {
+        YamlShardingRuleConfiguration result = convertTableRuleSegments(sqlStatement.getTables());
+        for (Collection<String> each : sqlStatement.getBindingTables()) {
+            result.getBindingTables().add(Joiner.on(",").join(each));
+        }
+        return result;
+    }
+    
+    /**
+     * Convert create sharding table rule statement context to YAML sharding rule configuration.
+     * 
+     * @param sqlStatement create sharding table rule statement
+     * @return YAML sharding rule configuration
+     */
+    public static YamlShardingRuleConfiguration convert(final CreateShardingTableRuleStatement sqlStatement) {
+        return convertTableRuleSegments(sqlStatement.getTables());
+    }
+    
+    private static YamlShardingRuleConfiguration convertTableRuleSegments(final Collection<TableRuleSegment> tableRuleSegments) {
         YamlShardingRuleConfiguration result = new YamlShardingRuleConfiguration();
-        for (TableRuleSegment each : sqlStatement.getTables()) {
+        for (TableRuleSegment each : tableRuleSegments) {
             if (null != each.getTableStrategy()) {
                 result.getShardingAlgorithms().put(getAlgorithmName(each.getLogicTable(), each.getTableStrategy().getAlgorithmName()), createAlgorithmConfiguration(each.getTableStrategy()));
                 result.getAutoTables().put(each.getLogicTable(), createAutoTableRuleConfiguration(each));
             }
         }
-        for (Collection<String> each : sqlStatement.getBindingTables()) {
-            result.getBindingTables().add(Joiner.on(",").join(each));
-        }
         return result;
     }
     
+    
```

Comment:
Is this a reductant line? I assume the blank line above is  enough.

Reply:
deleted